### PR TITLE
chore(github): pull request template

### PR DESCRIPTION
**Pull request by gurinderu**

### CODE REVIEW

**Review by justprosh — APPROVED** [review]

**Review by folex — APPROVED** [review]

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+- [ ] I have added/updated necessary comments to aid understanding.
+
+## Screenshots (if applicable)
+[Add any relevant screenshots or animated GIFs to showcase the changes.]
+
+## Additional Notes
+[Provide any additional information or context that may be helpful for the reviewer.]
+
+## Reviewer Checklist
+- [ ] Code has been reviewed for quality and adherence to guidelines (https://doc.rust-lang.org/1.0.0/style/README.html).
```
Comment (nahsi) [inline .github/pull_request_template.md:28]: Can we make a check out of it? Or maybe we already have (clippy?)

Comment (gurinderu) [reply .github/pull_request_template.md:28]: not sure about exact rules, but clippy should cover most of them

Comment (kmd-fl) [inline .github/pull_request_template.md:28]:
```suggestion
- [ ] Code has been reviewed for quality and adherence to [guidelines](https://doc.rust-lang.org/1.0.0/style/README.html).
```
?

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+- [ ] Documentation has been updated to reflect the changes (if applicable).
+- [ ] All new and existing unit tests have passed.
+- [ ] I have self-reviewed my code and ensured its quality.
+- [ ] I have added/updated necessary comments to aid understanding.
+
+## Screenshots (if applicable)
+[Add any relevant screenshots or animated GIFs to showcase the changes.]
+
+## Additional Notes
+[Provide any additional information or context that may be helpful for the reviewer.]
```
Comment (kmd-fl) [inline .github/pull_request_template.md:25]:
```suggestion
## Screenshots (if applicable)
[Add any relevant screenshots or animated GIFs to showcase the changes.]

## Additional Notes
[Provide any additional information or context that may be helpful for the reviewer.]

## Checklist
- [ ] The code follows the project's coding conventions and style guidelines.
- [ ] All tests related to the changes have passed successfully.
- [ ] Documentation has been updated to reflect the changes (if applicable).
- [ ] All new and existing unit tests have passed.
- [ ] I have self-reviewed my code and ensured its quality.
- [ ] I have added/updated necessary comments to aid understanding.
```

?

**Review by kmd-fl — APPROVED** [review]

**Review by folex — APPROVED** [review]

**Pull request merged**